doc/installing-a-development-environment.md

```diff
@@ -1,5 +1,17 @@
 ## Installing
 
+### Introduction to Installation
+
+The development environment setup requires installation of following softwares:
+
+1. Ruby - Needs to be installed in order to install and use Jekyll. Jekyll helps in tranforming plain texts into static websites.
+
+2. Python - Needs to be installed in order to invoke simple HTTP server from Makefile.
```

Comment:
If this is only needed for the "simple HTTP server from Makefile" then this is also optional (and should be moved down, after "Make")